lib/cdo/contact_rollups/v2/pardot.rb

```diff
@@ -270,7 +270,7 @@ def self.delete_prospect_by_id(pardot_id)
   # @param email [String]
   # @return [Array<String>]
   def self.retrieve_pardot_ids_by_email(email)
-    doc = post_with_auth_retry "#{PROSPECT_READ_URL}/#{URI.escape(email)}"
+    doc = post_with_auth_retry "#{PROSPECT_READ_URL}/#{URI.encode_www_form_component(email)}"
```

Comment:
Note that `URI.escape` does not correctly escape email addresses. In particular, it doesn't escape `@`:

```ruby
[development] dashboard > URI.escape("test@domain.com")
=> "test@domain.com"
[development] dashboard > CGI.escape("test@domain.com")
=> "test%40domain.com"
[development] dashboard > URI.encode_www_form_component("test@domain.com")
=> "test%40domain.com"
```

That means this does represent a definite change in behavior, but (I think) a desirable one.

Reply:
Just tested this out manually, and I can confirm that `%40` does indeed work! Since all of the recommended encoding alternatives encode `@` this way - and because regardless of salesforce's implementation, using this character is urls is not recommended - I still think this is a desirable change

Reply:
Just tested this out manually, and I can confirm that `%40` does indeed work! Since all of the recommended encoding alternatives encode `@` this way - and because regardless of salesforce's implementation, using this character is urls is not recommended - I still think this is a desirable change

Reply:
SGTM!